src/Core/Email/EmailLister.php

```diff
@@ -49,31 +50,30 @@ public function __construct(FileSystem $fs)
      */
     public function getAvailableMails($dir)
     {
-        if (!is_dir($dir)) {
+        try {
+            $mail_directory = $this->filesystem->listFilesRecursively($dir);
+        } catch (Exception $e) {
```

Comment:
Exceptions don't have to extend `\Exception` class, but they have to implement `\Throwable` interface at least.
```suggestion
        } catch (Throwable $e) {
```

Reply:
Thanks for the review and suggestion, but in this case, Exception is 'PrestaShop\PrestaShop\Core\Foundation\Filesystem\Exception' and not `\Exception`.

My intention is to catch only the possible types of exceptions that `FileSystem::listEntriesRecursively()` throws.

Reply:
I see in PHPStorm

<img width="533" alt="image" src="https://user-images.githubusercontent.com/194784/156452118-c50cfa78-70c0-4d35-aff8-d0a3fa31bf11.png">

I think this warning is caused by the fact that the function is not properly documented, @throws is missing in the PHPDoc section.

Reply:
No need of phpdoc. Static analysis in phpStorm is supposed to detect potential exceptions thrown by called method.
Perhaps code indexation wasn't completed when you took the screenshot ?
Here the warning from my own IDE:
![Capture d’écran de 2022-03-03 15-17-24](https://user-images.githubusercontent.com/97030953/156583305-a13494ae-73c8-4202-be71-bcf57ad6faa3.png)